docs/sdk/v3/guides/liquidity/01-minting-a-position.md

```diff
@@ -3,4 +3,90 @@ id: minting
+
+The first step is to give approval to the protocol's `NonfungiblePositionManager` to transfer our tokens. 
+
+```js reference title="Approving our tokens for transferring" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/6fba6da4d323804db56b3189ad1bbbaf18e6180f/v3-sdk/minting-position/src/example/Example.tsx#L113-L124
+```
+
+The logic to achieve that is wrapped in the `getTokenTransferApprovals` function. In short, since both **USDC** and **DAI** are ERC20 tokens, we setup a reference to their smart contracts and call the `approve` function:
+
+```js reference title="Setting up an ERC20 contract reference and approving" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/6fba6da4d323804db56b3189ad1bbbaf18e6180f/v3-sdk/minting-position/src/libs/contracts.ts#L73-L78
+```
+
+### Fetching the Pool's constant and state to create an instance of a `Pool` class
+
+Having approved the transfer of our tokens, we now need to get data about the pool for which we will provide liquidity, in order to instantiate a Pool class:
+
+```js reference title="Fetching the Pool's constants and current state" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/6fba6da4d323804db56b3189ad1bbbaf18e6180f/v3-sdk/minting-position/src/example/Example.tsx#L134-L135
+```
+
+:::info
+
+For a briefer on how to instantiate a reference to the Uniswap V3 Pool smart contract, and to retrieve the pool constants, please take a look at the [quoting guide](../quoting#setting-up-a-reference-to-the-pool-contract-and-getting-metadata-from-it)!
+
+:::
+
+Like we did for the Pool constants, we get the Pool's current state data by creating a reference to the Pool's smart contract and accessing its methods:
+
+```js reference title="Setting up a Pool contract reference and fetching current state data" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/74621ce380dec537a3f9654ec8723cc4be9e54b8/v3-sdk/minting-position/src/example/Example.tsx#L81-L86
+```
+
+Having collected the required data, we can now create an instance of the Pool helper class:
+
+```js reference title="Create a Pool representation instance " referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/74621ce380dec537a3f9654ec8723cc4be9e54b8/v3-sdk/minting-position/src/example/Example.tsx#L127-L134
+```
+
+### Creating an instance of a `Position` class and using `addCallParameters` to get the data for the transaction
+
+Having created the instance of the Pool class, we can now use that to create an instance of a Position class, which represents the position we want to create:
+
+```js reference title="Create a Position representation instance" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/74621ce380dec537a3f9654ec8723cc4be9e54b8/v3-sdk/minting-position/src/example/Example.tsx#L137-L146
+```
```

Comment:
Here is where we should explain how much liquidity you're providing / what `tickUpper` and  `tickLower` are doing. I still have no idea how much currency I am spending if you told me that liquidity = X value, and I should understand that while creating the position